### PR TITLE
Issue140/movingemails

### LINKED ISSUE #140: Moving emails to different folders
As a user I would like to move an email from folder A to folder B. So the email should not appear in folder A anymore.

### PR DESCRIPTION
**Pull request by mariwalt**
Closes #140.

### CODE REVIEW

File: packages/IMAPClient-UI.package/ICFolderDialog.class/instance/emailListMenu..st [hunk 1]
```diff
@@ -5,11 +5,12 @@ emailListMenu: aMenu
...
 		add: ((self selectedEmail retrieveFlags at: #flagged) 
 			ifTrue: ['Unflag']
 			ifFalse: ['Mark as flagged'])  
-			action: #modifyFlaggedFlag;
+			action: [self modifyFlag: #flagged];
 		add: (self labelForDeletionAction: self selectedEmail folder name)
-			action: #deleteEmail.	
+			action: #deleteEmail;
+		add: 'Move to ...' subMenu: self folderListMenu.	
 	^aMenu
```
Comment (mariwalt) [inline packages/IMAPClient-UI.package/ICFolderDialog.class/instance/emailListMenu..st]: small refactoring that is not really part of this issue, but it is nice to have:)

File: packages/IMAPClient-UI.package/ICFolderDialog.class/instance/modifyFlaggedFlag.st [hunk 2]
```diff
@@ -1,7 +0,0 @@
-flags
-modifyFlaggedFlag
-
-	selectedEmail modifyFlag: #flagged.
-	selectedEmail flagged: (selectedEmail retrieveFlags at: #flagged).
-	selectedEmail saveInMaildir.
-	self changed: #emailsOfSelectedFolder
```
Comment (mariwalt) [inline packages/IMAPClient-UI.package/ICFolderDialog.class/instance/modifyFlaggedFlag.st]: small refactoring that is not really part of this issue, but it is nice to have:)

File: packages/IMAPClient-UI.package/ICFolderDialog.class/instance/modifySeenFlag.st [hunk 3]
```diff
@@ -1,7 +0,0 @@
-flags
-modifySeenFlag
-
-	selectedEmail modifyFlag: #seen.
-	selectedEmail seen: (selectedEmail retrieveFlags at: #seen).
-	selectedEmail saveInMaildir.
-	self changed: #emailsOfSelectedFolder
```
Comment (mariwalt) [inline packages/IMAPClient-UI.package/ICFolderDialog.class/instance/modifySeenFlag.st]: small refactoring that is not really part of this issue, but it is nice to have:)

**Review by ClayGee — APPROVED** [review]
Seems to be working great :).

**Comment by davidhennemann** [comment]
So moving worked fine :D cool! But the List of Folders, was only targeting child folders on the first level. But on my Tonline Account e.g. all folders are a subfolder of the inbox. So the Move Menu only listed `INBOX`. 
So i added a simple BFS to the Folder Tree, which will navigate and collection folders through all child folders. But still not very optimal. 
There is a option to add submenus to a menu. So we should think about adding a dynamically generated submenu for each folder (which contains the children of each folder). I just added the Items retrieved from the BFS and add a few spaces for first level children. But i am not sure, whether we should create a new issue for "full" child-folder support, or include it in this one...What do you think? IMO it would take quite some time to correctly build a dynamic context menu with correct subfolders (and subfolders in them - if there are any)

image

**Review by davidhennemann — COMMENTED** [review]
So moving worked fine :D cool! But the List of Folders, was only targeting child folders on the first level. But on my Tonline Account e.g. all folders are a subfolder of the inbox. So the Move Menu only listed `INBOX`. 
So i added a simple DFS to the Folder Tree, which will navigate and collection folders through all child folders. But still not very optimal. 
There is a option to add submenus to a menu. So we should think about adding a dynamically generated submenu for each folder (which contains the children of each folder). I just added the Items retrieved from the BFS and add a few spaces for first level children. But i am not sure, whether we should create a new issue for "full" child-folder support, or include it in this one...What do you think? IMO it would take quite some time to correctly build a dynamic context menu with correct subfolders (and subfolders in them - if there are any)

image

**Comment by davidhennemann** [comment]
Do we still need the `Move to trash`? We can do this with the new context menu, can't we?

**Comment by mariwalt** [comment]
?I would keep, but perhaps rename it "Delete".

________________________________

**Pull request merged**